File: stubs/java/src/main/java/com/beastmode/rmddd/examples/ecommerce/OrderConfirmedEvent.java
package com.beastmode.rmddd.examples.ecommerce;

import com.beastmode.rmddd.domain.DomainEvent;

import java.math.BigDecimal;
import java.util.Map;

/**
 * Domain event fired when an order is confirmed.
 */
public class OrderConfirmedEvent extends DomainEvent {
    
    private final BigDecimal totalAmount;
    
    public OrderConfirmedEvent(OrderId orderId, BigDecimal totalAmount) {
        super(orderId);
        this.totalAmount = totalAmount;
    }
    
    public OrderId getOrderId() {
        return (OrderId) getAggregateId();
    }
    
    public BigDecimal getTotalAmount() {
        return totalAmount;
    }
    
    @Override
    public Map<String, Object> getEventData() {
        return Map.of(
            "order_id", getOrderId().getValue().toString(),
            "total_amount", totalAmount.toString()
        );
    }
}

File: stubs/csharp/src/BeastMode.RmDdd.Examples/Examples/ECommerce/Events/OrderConfirmedEvent.cs
using System.Collections.Generic;
using BeastMode.RmDdd.Domain;

namespace BeastMode.RmDdd.Examples.ECommerce.Events
{
    /// <summary>
    /// Domain event fired when an order is confirmed.
    /// </summary>
    public class OrderConfirmedEvent : DomainEvent
    {
        public decimal TotalAmount { get; }
        
        public OrderConfirmedEvent(OrderId orderId, decimal totalAmount) : base(orderId)
        {
            TotalAmount = totalAmount;
        }
        
        public OrderId OrderId => (OrderId)AggregateId;
        
        public override IDictionary<string, object> GetEventData()
        {
            return new Dictionary<string, object>
            {
                ["order_id"] = OrderId.Value.ToString(),
                ["total_amount"] = TotalAmount
            };
        }
    }
}